lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp

```diff
@@ -697,31 +703,45 @@ namespace Js
 
         return retVal;
     }
-    Var IntlEngineInterfaceExtensionObject::EntryIntl_ResolveLocaleLookup(RecyclableObject* function, CallInfo callInfo, ...)
+
+    Var IntlEngineInterfaceExtensionObject::EntryIntl_IsLocaleAvailable(RecyclableObject* function, CallInfo callInfo, ...)
     {
         EngineInterfaceObject_CommonFunctionProlog(function, callInfo);
 
+#ifdef INTL_ICU
         if (args.Info.Count < 2 || !JavascriptString::Is(args.Values[1]))
         {
-            // NormalizeLanguageTag of undefined or non-string is undefined
             return scriptContext->GetLibrary()->GetUndefined();
         }
 
-        JavascriptString *argString = JavascriptString::FromVar(args.Values[1]);
-        PCWSTR passedLocale = argString->GetSz();
+        JavascriptString *locale = JavascriptString::FromVar(args.Values[1]);
+        utf8::WideToNarrow locale8(locale->GetSz(), locale->GetLength());
 
-#if defined(INTL_ICU)
-        char16 resolvedLocaleName[ULOC_FULLNAME_CAPACITY] = { 0 };
-        if (ResolveLocaleLookup(passedLocale, resolvedLocaleName))
+        return TO_JSBOOL(scriptContext, IsLocaleAvailable(locale8));
+#else
+        // Only ICU supports this function
```

Comment:
>// Only ICU supports this function [](start = 8, length = 34)

Is returning null here a sentinel to JS to fallback (as I used previously) or something else?

Reply:
Sortof, but really, I am not expecting this code to ever be called from a WinGlob path. Should that be a failfast instead?